async-opcua-client/src/session/services/subscriptions/event_loop.rs

```diff
@@ -17,6 +17,11 @@ pub enum SubscriptionActivity {
     /// A publish request failed, either due to a timeout or an error.
     /// The publish request will typically be retried.
     PublishFailed(StatusCode),
+    /// Fatal failure, a publishing request has failed fatally in a way
+    /// that indicates it will not recover on its own.
+    /// This typically means the client has lost connection to the server.
+    /// When this is received by the session event loop it triggers a session restart.
```

Comment:
So does this change behavior for clients that also hook into the event_loop and have additional logic around reconnects?